aiohttp/web_routedef.py

```diff
@@ -192,3 +193,28 @@ def view(self, path: str, **kwargs: Any) -> _Deco:
     def static(self, prefix: str, path: PathLike,
                **kwargs: Any) -> None:
         self._items.append(StaticDef(prefix, path, kwargs))
+
+
+class ClassRouteTableDef(RouteTableDef):
+    def __repr__(self) -> str:
+        return "<ClassRouteTableDef count={}>".format(len(self._items))
+
+    def route(self,
+              method: str,
+              path: str,
+              **kwargs: Any) -> _Deco:
+        def inner(handler: Any) -> Any:
+            # Is Any rather than _HandlerType because of mypy limitations, see
+            # https://github.com/python/mypy/issues/3882 and
+            # https://github.com/python/mypy/issues/2087
+            handler.route_info = (method, path, kwargs)
+            return handler
+        return inner
+
+    def add_class_routes(self, instance: Any) -> None:
+        def predicate(member: Any) -> bool:
+            return all((inspect.iscoroutinefunction(member),
```

Comment:
```suggestion
            return all((inspect.ismethod(member),
                        hasattr(member, "route_info")))
        for instance in instances:
            for _, handler in inspect.getmembers(instance, predicate):
                assert inspect.iscoroutinefunction(handler), \
                    "Please ensure your decorated handler methods are all \"async\""
                method, path, kwargs = handler.route_info
                super().route(method, path, **kwargs)(handler)
```

Not sure if this will be helpful here but I refactored this a little in some personal code I have to catch cases where someone accidentally decorates a non-async function within a class.